Before & After callbacks also receive raw token

dakujem opened the pull request:
## What has been changed
Callbacks `after` and `before` receive both the decoded claims and the raw token string, as opposed to only receiving the decoded claims.

## The motivation
I needed to process/use the raw token further and the middleware does not offer any way to provide, as `fetchToken`'s visibility is _private_ and extension is not allowed.

I am forced to copy the very same code that fetches the token without this change.

## Compatibility
Backward & forward compatible, non-breaking change.

## Thanks
It would be awesome if this simple change could be swiftly integrated. Thank you in advance. 🥇 

## Implementation & Tests

1. I named the raw token argument as `raw`, I was also considering `encoded`, if you prefer the latter, let me know, I'll quickly update the PR.
2. I updated the test method `testShouldCallAfter` to also test for the added argument being passed correctly. Then I copied the code and created almost exacly the same test, only for the `before` callback - `testShouldCallBeforeWithProperArguments`. I copied the method for sake of more atomic tests. Though I'm not really fond of copying, so if you are like me, I can create a more convoluted test named for example `testShouldCallBeforeAndAfterWithProperArguments` that would test both cases.

Let me know what you think & thank you for cooperation and this nice middleware.

tuupola commented:
Quickly looking looks good to me. I will check it through later today. Perfect PR with test and everything. Thanks!

tuupola commented:
Been pondering naming a bit. I do like `raw` but I think `token` would be better. Could you do that change and I will merge.

dakujem commented:
of course, give me a minute...

dakujem committed:
Before & After callbacks also receive raw token

- `before` and `after` callbacks receive both the decoded claims and the raw token string

dakujem commented:
Okay, done. Please run the tests, as I was not able to.

dakujem commented:
Mika, I also added a note to the docs mentioning this change, as well as fixed a typo.

tuupola commented:
Released as `3.4.0`. Thanks!

dakujem commented:
Thank _you_ for the swift cooperation!
